ci/check-nix-tools.nix

```diff
@@ -14,7 +14,10 @@ writeScript "check-nix-tools.sh" ''
 
   export PATH="${makeBinPath [ stdenv.shellPackage coreutils nixStable git gawk ]}:$PATH"
 
-  cd $(git rev-parse --show-toplevel)
+  if [ -z "''${BUILDKITE:-}" ]; then
+    # Go to top of project repo, unless running under CI.
+    cd $(git rev-parse --show-toplevel)
+  fi
 
   # The regenerate script is here by convention
   ./nix/regenerate.sh
```

Comment:
not part of the changes if this PR, but maybe it would be better to use `nix-tools.regeneratePackages` directly to avoid relying on a convention?